### PR TITLE
docs: Fix minor typos and grammar

### PR DESCRIPTION
**Pull request by spydon**
## What kind of change does this PR introduce?

Just some minor fixes of typos, grammar and broken links.
If you don't want modifications to the `changelog` I can revert those.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,13 +65,13 @@
   ```
   - `auth.signIn()` is split into different methods
   ```dart
-    // Magic link signin
+    // Magic sign-in link
```
Comment (dshukertjr) [inline CHANGELOG.md:68]: `Magic link` is the term we typically refer to this feature as Supabase, so could we do this?

```suggestion
    // Magic link sign-in
```

File: example/macos/Flutter/GeneratedPluginRegistrant.swift [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import FlutterMacOS
 import Foundation
 
-import app_links_macos
+import app_links
```
Comment (dshukertjr) [inline example/macos/Flutter/GeneratedPluginRegistrant.swift:8]: Do we need these changes on the generated files for macos and windows?

**Review by dshukertjr — COMMENTED** [review]
Thanks, this is great! I think the fixes on the changelog is fine too!

File: example/macos/Flutter/GeneratedPluginRegistrant.swift (hunk 2 above)
Comment (spydon) [reply example/macos/Flutter/GeneratedPluginRegistrant.swift:8]: That was accidentally committed, reverting that!

Comment (spydon) [reply example/macos/Flutter/GeneratedPluginRegistrant.swift]: Done

**Review by dshukertjr — APPROVED** [review]
Looks amazing! Thank you so much for finding and correcting all these small mistakes!

**Pull request merged**